### PR TITLE
Add tests for add and remove

### PR DESCRIPTION
**Pull request by billymohajeri**
In this step we,
- Used jest framework for testing.
- Created a test file (crud.test.js).
- Created multiple tests for the add function.
- Created multiple tests for the remove function.
- Added `--coverage` flag to our test script in package.json.

### CODE REVIEW

File: src/crud.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    addToArr('newTask3', arr);
+    expect(arr[3].index).toBe(4);
+  });
+
+  test('Adding a new element at 4', () => {
+    addToArr('New text', arr);
+    expect(arr[4].description).toBe('New text');
+  });
+  test('Adding a new element at 5', () => {
+    addToArr('newTask5', arr);
```
Comment (alexander16108) [inline src/crud.test.js:26]: Hi there 👋🏾  your test looks pretty good and detailed but as required you  still need to create a mock  of the local storage, DOM or External API of your web app

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @billymohajeri , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
your doing soo good just a little more to add the icing on the cake



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/crud.test.js (hunk 1 above)
Comment (billymohajeri) [reply src/crud.test.js:26]: Hi @alexander16108 and thank you for the feedback.
As you know this is **Testing To Do list: part 1** and the requirement is to write tests for the "add" and "remove" functions.

Here in the [requirements ](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md)says that:
> Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be straightforward. Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks

So as our "add" and "remove" functions were pure functions we wrote the tests like that.
Obviously in the **Testing To Do list: part 2** we have to write mock tests of the local storage, DOM.

Please correct me if I'm wrong.
Thanks

**Review by Lameck1 — COMMENTED** [review]
Hello @billymohajeri 👋,

_This is an additional review as a result of the second opinion request._

Allow me to explain the project requirement you are trying to highlight.

>Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks:
> - Take a look at each of these functions. Are they *pure functions*? If the answer is "yes" then writing tests for them should be straightforward. Some of those functions however will update *localStorage* and manipulate the *DOM*. For those, you will need to use mocks:
>   - Mock a storage object to "imitate" localStorage operations (only if you are using `jest < 24.x`, in newer versions localStorage is mocked automatically).
>   - Mock HTML to test if add/delete functions add or remove exactly one `<li>` element to/from the list in the DOM.

The requirement highlighted above is two-part and neither part undoes the other. What it is trying to communicate is that your functions that are responsible for adding and removing a task might either be:

1. **_PURE FUNCTIONS together with other FUNCTIONS that separately take care of updating the localStorage_**. In this case, test the pure function as they are but still go ahead and test the ones that update the localStorage by writing mocks.

2. **_NON-PURE FUNCTIONS that directly work with and update the localStorage_**. In this case, you will have to mock the localStorage first before testing these functions.

Testing should be proof of the functionality of your app. Have you covered the necessary edge cases? Our app is expected to take an input, save it in the localStorage, and then display it in the UI. This is what the requirement quoted above is looking for. You'll have to mock both the HTML and the localStorage to realize this.

Please work on the concerns raised by the CR and then ask for another code review.

Keep rocking 🚀!

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**